nixos/modules/services/networking/prosody.nix

```diff
@@ -1,885 +1,659 @@
-{ config, lib, pkgs, ... }:
-
-with lib;
+{ options, config, pkgs, lib, ... }:
 let
-  cfg = config.services.prosody;
+  inherit (lib) mkChangedOptionModule mkDefault mkEnableOption mkForce mkIf mkMerge mkOption mkRenamedOptionModule;
+  inherit (lib) all any attrValues boolToString concatStringsSep filterAttrs genAttrs hasPrefix listToAttrs literalExpression mapAttrsToList mdDoc optionals optionalAttrs optionalString types unique;
 
-  sslOpts = { ... }: {
+  config' = config;
```

Comment:
This is so that we can access the top-level config inside of submodules? Maybe we can create cfgX instead? I didn't look into this compeltely.

Reply:
Yes that is why. What is `cfgX` denoting that `config'` isn't? I think these are just two conventions?